### PR TITLE
#631 optional password in user form

### LINKED ISSUE #631: Make password optional in create user form
### What needs to be done
 
image

Make the password optional, so its easier for the admin to just use a reset link

### PR DESCRIPTION
**Pull request by danseremet**
### Test plan
Unit test passes.
Demo:
PasswordOptional

##### _List all issues this PR resolves_
- closes #631 

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [x] Documentation

### CODE REVIEW

File: app/Http/Controllers/Admin/UserController.php [hunk 1]
```diff
@@ -44,16 +44,28 @@ public function index()
...
+                'name' => ['required', 'string', self::MAX_255],
+                'email' => ['required', 'string', 'email', self::MAX_255, 'unique:users']
+            ]);
+        } else {
+            $request->validateWithBag('createUser', [
+                'name' => ['required', 'string', self::MAX_255],
+                'email' => ['required', 'string', 'email', self::MAX_255, 'unique:users'],
+                'password' => ['required', 'string', self::MAX_255, 'confirmed', 'min:8'],
+            ]);
+        }
```
Comment (a-drew) [inline app/Http/Controllers/Admin/UserController.php:63]:
```suggestion
        $request->validateWithBag('createUser', [
            'name' => ['required', 'string', self::MAX_255],
            'email' => ['required', 'string', 'email', self::MAX_255, 'unique:users'],
            'password' => ['string', self::MAX_255, 'confirmed', 'min:8'],
        ]);
        
        // Randomly creates a temporary password when it is not set.
        // The user can login using a reset link.
        $password = $request->password;
        if (!$password) {
            $password = Str::random(20);
        }
```
no need to duplicate validation when you can remove the require rule

**Review by a-drew — DISMISSED** [review]
Looks good, I left a suggestion to improve it a bit

File: app/Http/Controllers/Admin/UserController.php (hunk 1 above)
Comment (danseremet) [reply app/Http/Controllers/Admin/UserController.php:63]: Does the 'min:8' rule not apply when there is no 'required' rule? Does it only apply when there is at least a character?

Comment (danseremet) [reply app/Http/Controllers/Admin/UserController.php]: It still applies, it also applies on 'string', this logic is not the right behavior, I will revert this change.

**Review by a-drew — APPROVED** [review]
take2interactive

File: app/Http/Controllers/Admin/UserController.php (hunk 1 above)
Comment (a-drew) [reply app/Http/Controllers/Admin/UserController.php:64]: odd but this isn't a super important change. let's merge

**Review by alexstojda — APPROVED** [review]
LGTM!

**Pull request merged**